From 19eff8241db97fa88ff8c56af35f2f8820c0f2bb Mon Sep 17 00:00:00 2001
From: Arne Coucheron <arco68@gmail.com>
Date: Mon, 22 Feb 2016 06:50:53 +0100
Subject: [PATCH 063/172] Snap: Remove face detection option if not supported

Removes the face detection option from preference list if
the "face-detection" camera parameter is blank/doesn't exist.

Change-Id: I2ec82ae38505b6674f1a34e98d1b1fb2dbc2a714
---
 src/com/android/camera/CameraSettings.java | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/com/android/camera/CameraSettings.java b/src/com/android/camera/CameraSettings.java
index 6b6a2f9ca..86eede661 100755
--- a/src/com/android/camera/CameraSettings.java
+++ b/src/com/android/camera/CameraSettings.java
@@ -157,6 +157,7 @@ public class CameraSettings {
     private static final String KEY_QC_SUPPORTED_VIDEO_CDS_MODES = "video-cds-mode-values";
     private static final String KEY_QC_SUPPORTED_TNR_MODES = "tnr-mode-values";
     private static final String KEY_QC_SUPPORTED_VIDEO_TNR_MODES = "video-tnr-mode-values";
+    private static final String KEY_QC_SUPPORTED_FACE_DETECTION = "face-detection-values";
     private static final String KEY_SNAPCAM_SUPPORTED_HDR_MODES = "hdr-mode-values";
     private static final String KEY_SNAPCAM_SUPPORTED_HDR_NEED_1X = "hdr-need-1x-values";
     public static final String KEY_SNAPCAM_SHUTTER_SPEED = "shutter-speed";
@@ -583,6 +584,14 @@ public class CameraSettings {
         return split(str);
     }
 
+    public static List<String> getSupportedFaceDetection(Parameters params) {
+        String str = params.get(KEY_QC_SUPPORTED_FACE_DETECTION);
+        if (str == null) {
+            return null;
+        }
+        return split(str);
+    }
+
     public static List<String> getSupportedDISModes(Parameters params) {
         String str = params.get(KEY_QC_SUPPORTED_DIS_MODES);
         if (str == null) {
@@ -994,6 +1003,11 @@ public class CameraSettings {
                     faceRC, getSupportedFaceRecognitionModes(mParameters));
         }
 
+        if (faceDetection != null) {
+            filterUnsupportedOptions(group,
+                    faceDetection, getSupportedFaceDetection(mParameters));
+        }
+
         if (autoExposure != null) {
             filterUnsupportedOptions(group,
                     autoExposure, ParametersWrapper.getSupportedAutoexposure(mParameters));
-- 
2.17.1

